### ISSUE TITLE
$PIPE lookup broken?

### ISSUE THREAD

**Issue by jpmens** [ansible/ansible#1729]
I think the PIPE plugin is broken in the latest devel. 

Version: `ansible 1.0 (devel 11b64314df) last updated 2012/12/05 15:46:35 (GMT +200)`
## Playbook

``` yaml

---
- hosts: 127.0.0.1
  connection: local
  gather_facts: false
  vars:
  - pcont: $PIPE(/bin/ls)
  tasks:
    - action: command echo $pcont

```
## run

``` sh
ansible-playbook listpipe.yml



PLAY [127.0.0.1] ********************* 
Traceback (most recent call last):
  File "/Users/jpm/Auto/pubgit/ansible/ansible/bin/ansible-playbook", line 177, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/Users/jpm/Auto/pubgit/ansible/ansible/bin/ansible-playbook", line 148, in main
    pb.run()
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/playbook/__init__.py", line 215, in run
    if not self._run_play(play):
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/playbook/__init__.py", line 428, in _run_play
    if not self._run_task(play, task, False):
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/playbook/__init__.py", line 292, in _run_task
    self.callbacks.on_task_start(utils.template(play.basedir, task.name, task.module_vars), is_handler)
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/utils/template.py", line 240, in template
    text = varReplace(basedir, unicode(text), vars, expand_lists=expand_lists)
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/utils/template.py", line 198, in varReplace
    replacement = varReplace(basedir, replacement, vars, depth=depth+1, expand_lists=expand_lists)
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/utils/template.py", line 186, in varReplace
    m = _varFind(basedir, raw, vars, depth)
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/utils/template.py", line 164, in _varFind
    replacement = instance.run(args, inject=vars)
  File "/Users/jpm/Auto/pubgit/ansible/ansible/lib/ansible/runner/lookup_plugins/pipe.py", line 27, in run
    p = subprocess.Popen(terms, cwd=self.basedir, shell=True, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/subprocess.py", line 595, in __init__
    errread, errwrite)
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/subprocess.py", line 1106, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory: ''
```

FWIW, `test/lookup_plugins.yml` fails similarly.

**Comment by jpmens** [comment]
`$FILE()` and `$DNSTXT()` work fine.

**Comment by dhozac** [comment]
Easy fix, have it pending #1733 being merged. You can work around it by using ./playbook.yml...
